### DISCUSSION TITLE
dFBA

### DISCUSSION THREAD

**Discussion by FH96** [opencobra/cobrapy#1400] [General]
In the dFBA implementation in the docs I noticed glucose is maximized when added as lexicographic constraint. Since a negative flux indicates glucose uptake I was wondering if maximizing glucose uptake, actually means pushing organism to use more glucose or less?

**Comment by cdiener** [comment]
Yes, you are correct. The second optimization minimizes the glucose uptake, so it pushes the model to use the least amount of glucose necessary to reach the maximum growth rate. We should probably also have limited the upper bound for the reaction to 0 to avoid glucose production for better clarity.

**Reply by FH96** [reply]
Thank you for the clarification, it seems to be an important thing to consider when ordering lexicographic constraints.

Also, when I run dFBA sometimes it gets infeasible and gives an error, while other times, when it's infeasible, it finishes earlier than the specified time without any error. I'd appreciate any insight on this.

**Reply by cdiener** [reply]
I could imagine this has to do with non-unique fluxes. Because in each FBA step only the optimized quantities have a unique solution, but most fluxes don't, there can be issue at the update steps. The step size will also influence this to a degree. 

You could check out the [dfba package from the Touirigny group](https://gitlab.com/davidtourigny/dynamic-fba) which solves those issues.
